core/flyout_base.js

```diff
@@ -457,8 +457,8 @@ Blockly.Flyout.prototype.hide = function() {
 
 /**
  * Show and populate the flyout.
- * @param {!Blockly.utils.toolbox.FlyoutDefinition} flyoutDef Contents to display
- *     in the flyout. This is either an array of Nodes, a Node, a
+ * @param {!Blockly.utils.toolbox.FlyoutDefinition|string} flyoutDef Contents to display
```

Comment:
Why add a string as part of the type here, instead of making `FlyoutDefinition` include `string`?

Reply:
Ahh, I found the culprit. So, `convertFlyoutDefToJsonArray` actually does not take in a string. We handle any custom categories before hand in the show method using `getDynamicCategoryContents_` before passing to the `convertFlyoutDefToJsonArray`. We could move the `getDynamicCategoryContents_` to the utils file, we would just need to pass in the workspace anywhere we call `convertFlyoutDefToJsonArray`.

Reply:
:+1: 

No opinion on whether to move it into utils.

Reply:
Ok planning on leaving it for now then.